src/algebra/euclidean_domain.lean

```diff
@@ -230,6 +230,78 @@ by haveI := classical.dec_eq α; exact
 
 end gcd
 
+section lcm
+variables [decidable_eq α]
+
+def lcm (x y : α) : α :=
+if gcd x y = 0 then 0 else x * y / gcd x y
```

Comment:
Isn't the if statement unnecessary? I thought division was defined to work like this

Reply:
No, 0/0 is 1.

Reply:
If that's true, then how can `(/)` be the quotient argument for `euclidean_domain int`? I'm pretty sure that `0 / 0 : int` is `0`.

Reply:
We're both wrong: 0/0 is undefined. See zulip.

Reply:
done